### PR TITLE
Cleanup/codebase improvement

### PR DESCRIPTION
**Pull request by slakh96**
Completes https://github.com/orgs/VerWiki/projects/1#card-57106716 and https://github.com/orgs/VerWiki/projects/1#card-56837905

### CODE REVIEW

File: frontend/src/components/Tree/Tree.jsx [hunk 1]
```diff
@@ -95,12 +96,14 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick, onRightClick) {
...
       (d) => `
-        rotate(${(d.x * 180) / Math.PI - 90})
+        rotate(${(d.x * TreeConf.NODE_ROTATE_FACTOR) / Math.PI - 90})
         translate(${d.y},0)
       `
     )
-    .attr("fill", (d) => (d.data.numChildren === 0 ? "#b30000" : "#555"))
-    .attr("r", 6)
+    .attr("fill", (d) =>
+      d.data.numChildren === 0 ? TreeConf.LEAF_COLOUR : TreeConf.NODE_COLOUR
```
Comment (rafiiary) [inline frontend/src/components/Tree/Tree.jsx]: Just leaving this here as a way to remind you to change the LEAF_COLOR and NODE_COLOR!

File: frontend/src/utils/config.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Config values for the entire application. Ensure to document each constant!
+// See README.md for more details.
+
+//TreeViewer.jsx config
+export const VConf = Object.freeze({
```
Comment (rafiiary) [inline frontend/src/utils/config.js:5]: Can we find a better name for this? Like vConst since they're supposed to be constants?

Comment (slakh96) [reply frontend/src/utils/config.js:5]: VConf stands for viewerConf, maybe I should make it tvconf? so it is more clear that this config refers to tree viewer?

Comment (rafiiary) [reply frontend/src/utils/config.js:5]: Okay that's fine then. tvconf is definitely more confusing so just leave it haha

File: frontend/src/components/Tree/Tree.jsx (hunk 1 above)
Comment (slakh96) [reply frontend/src/components/Tree/Tree.jsx]: Done! Lmk if any issues with the colours

File: frontend/src/components/TreeViewer/TreeViewer.jsx [hunk 3]
```diff
@@ -235,18 +240,32 @@ function toggleInfoBoxVisibility(clickedNodeName, curViewingNodeName) {
   return nodeViewingAfterToggle;
 }
 
-export const TreeViewer = ({ data, treeID }) => {
+/**
+ * Gets the arguments of the query string from a given URL
+ * @param {string} name The name of the query-string parameter to get
+ * @param {string} url The url from where to parse query string
+ */
+const getParameterByName = (name, url) => {
```
Comment (slakh96) [inline frontend/src/components/TreeViewer/TreeViewer.jsx:248]: From @mit1221 : "We should probably add the base address here ("http://localhost:3003") and all other uses of it to the config file" Source: https://github.com/VerWiki/VerWikiApp/pull/22#discussion_r611084902

File: frontend/src/components/TreeViewer/TreeViewer.jsx [hunk 4]
```diff
@@ -235,18 +240,32 @@ function toggleInfoBoxVisibility(clickedNodeName, curViewingNodeName) {
...
+  name = name.replace(/[[\]]/g, "\\$&");
```
Comment (slakh96) [inline frontend/src/components/TreeViewer/TreeViewer.jsx:249]: From @mit1221 : "There already exist functions in JS for doing this: https://www.sitepoint.com/get-url-parameters-with-javascript/. I think it's safer to use this instead." Source: https://github.com/VerWiki/VerWikiApp/pull/22#discussion_r611085214

File: frontend/src/components/TreeViewer/TreeViewer.jsx (hunk 3 above)
Comment (slakh96) [reply frontend/src/components/TreeViewer/TreeViewer.jsx:248]: Done

File: frontend/src/components/TreeViewer/TreeViewer.jsx (hunk 4 above)
Comment (slakh96) [reply frontend/src/components/TreeViewer/TreeViewer.jsx:249]: From Sultan: He tried to use the existing functions but they didn't exactly fit our needs; so he had to make a new one

File: backend/repopulate_db.py [hunk 5]
```diff
@@ -3,20 +3,16 @@
 # production.
```
Comment (mit1221) [inline backend/repopulate_db.py:3]: This file is getting quite big. I think we should make a `backend/data` folder and store .json files for each course and then repopulate_db.py should read and process all files in the `backend/data` folder.

Comment (slakh96) [reply backend/repopulate_db.py:3]: Resolved; we agreed to leave this for now as it is getting replaced in phase 2, and having some data in the json being modified in this file would make things confusing

**Review by mit1221 — APPROVED** [review]

**Pull request merged**